client/src/pages/SignUpModal.tsx

```diff
@@ -0,0 +1,201 @@
+            if (!copy.includes("at least 1 lowercase letter")){
+                copy.push("at least 1 lowercase letter");    
+            }
+            setRequirements(copy);
+        }else{
+            setHasOneLowerCase(true);
+            if (copy.includes("at least 1 lowercase letter")){
+                const index = copy.indexOf("at least 1 lowercase letter");
+                copy.splice(index, 1);  
+            }
+            setRequirements(copy);
+        }
+
+        if (!oneUpperCase.test(password)){
+            setHasOneUpperCase(false);
+            if (!copy.includes("at least 1 uppercase letter")){
+                copy.push("at least 1 uppercase letter");    
+            }
+            setRequirements(copy);
+        }else{
+            setHasOneUpperCase(true);
+            if (copy.includes("at least 1 uppercase letter")){
+                const index = copy.indexOf("at least 1 uppercase letter");
+                copy.splice(index, 1);  
+            }
+            setRequirements(copy);
+        }
+
+        if (!oneNumber.test(password)) {
+            setHasOneNumber(false);
+            if (!copy.includes("at least 1 number")){
+                copy.push("at least 1 number");    
+            }
+            setRequirements(copy);
+        }else{
+            setHasOneNumber(true);
+            if (copy.includes("at least 1 number")){
+                const index = copy.indexOf("at least 1 number");
+                copy.splice(index, 1);  
+            }
+            setRequirements(copy);
+        }
+     
+        if (!oneSymbol.test(password)){
+            setHasOneSymbol(false);
+            if (!copy.includes("at least 1 symbol")){
+                copy.push("at least 1 symbol");    
+            }
+            setRequirements(copy);
+        }else{
+            setHasOneSymbol(true);
+            if (copy.includes("at least 1 symbol")){
+                const index = copy.indexOf("at least 1 symbol");
+                copy.splice(index, 1);  
+            }
+            setRequirements(copy);
+        }
+
+        if (!twelveCharacterMin.test(password)){
+            setHasTwelveCharacterMin(false);
+            if (!copy.includes("12 characters minimum")){
+                copy.push("12 characters minimum");    
+            }
+            setRequirements(copy);
+        }else{
+            setHasTwelveCharacterMin(true);
+            if (copy.includes("12 characters minimum")){
+                const index = copy.indexOf("12 characters minimum");
+                copy.splice(index, 1);  
+            }
+            setRequirements(copy);
+        }
+        setPassword(e.target.value);
+    }
+
+    const modalTitle = "Create Your Account";
+    const subtitle = "Register your email and create a password";
+
+    const popover = (
+        <Popover id="popover-basic" show={!hasOneLowerCase || !hasOneUpperCase || !hasOneNumber || !hasOneSymbol || !hasTwelveCharacterMin}>
```

Comment:
I suggest abstracting `!hasOneLowerCase || !hasOneUpperCase || !hasOneNumber || !hasOneSymbol || !hasTwelveCharacterMin` because it is reused.

```
const requirementsAreFulfilled = () => !hasOneLowerCase || !hasOneUpperCase || !hasOneNumber || !hasOneSymbol || !hasTwelveCharacterMin

<Popover id="popover-basic" show={requirementsAreFulfilled()}>
```